board/piksiv3/rootfs-overlay/etc/init.d/usb_upgrade.sh

```diff
@@ -39,7 +40,19 @@ echo "Performing upgrade..."
 # Killing any processes that could hurt user experience
 /etc/init.d/S90monit stop
 /etc/init.d/S83zmq_adapter_rpmsg_piksi100 stop
+
+# Send output to show the user what we're doing.
+while true; do
+  sleep $LOG_INTERVAL
+  let ELAPSED=$ELAPSED+$LOG_INTERVAL
+  echo "Upgrade in progress: $ELAPSED seconds elapsed"
+done | sbp_log $LOGLEVEL &
+
 upgrade_tool --debug $FIRMWARE | sbp_log $LOGLEVEL
+
+# Stop user output
+kill `jobs -p`
```

Comment:
can you explain to me what jobs does?  is there a different way to store the PID of the while loop?  Is this guaranteed to kill only the while true; area and not something else in the background?